shared/stateutil/state_root.go

```
@@ -0,0 +1,223 @@
+		return [32]byte{}, errors.Wrap(err, "could not compute previous justified checkpoint merkleization")
+	}
+	fieldRoots[17] = prevCheckRoot[:]
+
+	// CurrentJustifiedCheckpoint data structure root.
+	currJustRoot, err := checkpointRoot(state.CurrentJustifiedCheckpoint)
+	if err != nil {
+		return [32]byte{}, errors.Wrap(err, "could not compute current justified checkpoint merkleization")
+	}
+	fieldRoots[18] = currJustRoot[:]
+
+	// FinalizedCheckpoint data structure root.
+	finalRoot, err := checkpointRoot(state.FinalizedCheckpoint)
+	if err != nil {
+		return [32]byte{}, errors.Wrap(err, "could not compute finalized checkpoint merkleization")
+	}
+	fieldRoots[19] = finalRoot[:]
+
+	root, err := bitwiseMerkleize(fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
+	if err != nil {
+		return [32]byte{}, errors.Wrap(err, "could not compute full beacon state merkleization")
+	}
+	return root, nil
+}
+
+func forkRoot(fork *pb.Fork) ([32]byte, error) {
+	fieldRoots := make([][]byte, 3)
+	prevRoot := bytesutil.ToBytes32(fork.PreviousVersion)
+	fieldRoots[0] = prevRoot[:]
+	currRoot := bytesutil.ToBytes32(fork.CurrentVersion)
+	fieldRoots[1] = currRoot[:]
+	forkEpochBuf := make([]byte, 8)
+	binary.LittleEndian.PutUint64(forkEpochBuf, fork.Epoch)
+	epochRoot := bytesutil.ToBytes32(forkEpochBuf)
+	fieldRoots[2] = epochRoot[:]
+	root, err := bitwiseMerkleize(fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
+	if err != nil {
+		return [32]byte{}, err
+	}
+	return root, nil
+}
+
+func checkpointRoot(checkpoint *ethpb.Checkpoint) ([32]byte, error) {
+	fieldRoots := make([][]byte, 2)
+	if checkpoint != nil {
+		epochBuf := make([]byte, 8)
+		binary.LittleEndian.PutUint64(epochBuf, checkpoint.Epoch)
+		epochRoot := bytesutil.ToBytes32(epochBuf)
+		fieldRoots[0] = epochRoot[:]
+		fieldRoots[1] = checkpoint.Root
+	}
+	root, err := bitwiseMerkleize(fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
+	if err != nil {
+		return [32]byte{}, nil
```

Can you return the err here?